modules/core/src/main/java/org/apache/ignite/internal/processors/cache/IgniteCacheOffheapManagerImpl.java

```diff
@@ -2018,8 +2108,17 @@ private boolean isIncrementalDrEnabled(GridCacheContext cctx) {
         {
             assert mvccVer != null || newMvccVer == null : newMvccVer;
 
-            if (!busyLock.enterBusy())
-                throw new NodeStoppingException("Operation has been cancelled (node is stopping).");
+            while (!busyLock.enterBusy()) {
```

Comment:
Let's consider non-blocking partition reconciliation finalization. Not sure that it'll work. In any case while look with Thread.sleep() looks as inappropriate solution.